cmd/snap/cmd_model.go

```diff
@@ -94,25 +102,44 @@ func init() {
 }
 
 func (x *cmdModel) Execute(args []string) error {
-	var assertion asserts.Assertion
-	var err error
-	if x.Serial {
-		assertion, err = x.client.CurrentSerialAssertion()
-	} else {
-		assertion, err = x.client.CurrentModelAssertion()
-	}
+	var mainAssertion asserts.Assertion
+	serialAssertion, serialErr := x.client.CurrentSerialAssertion()
+	modelAssertion, modelErr := x.client.CurrentModelAssertion()
 
-	if err != nil {
-		if client.IsAssertionNotFoundError(err) {
+	// if we didn't get a model assertion bail early
+	if modelErr != nil {
+		if client.IsAssertionNotFoundError(modelErr) {
 			// device is not registered yet - use specific error message
-			err = fmt.Errorf(i18n.G("device not ready - no assertion found"))
+			return errNoMainAssertion
 		}
-		return err
+		return modelErr
+	}
+
+	// if the serial assertion error is anything other than not found, also
+	// bail early
+	// the serial assertion not being found may not be fatal
+	if serialErr != nil && !client.IsAssertionNotFoundError(serialErr) {
+		return serialErr
+	}
+
+	if x.Serial {
+		mainAssertion = serialAssertion
+	} else {
+		mainAssertion = modelAssertion
 	}
 
-	// --assertion means output the raw assertion
 	if x.Assertion {
-		_, err = Stdout.Write(asserts.Encode(assertion))
+		if x.Verbose {
+			// can't do a verbose mode for the assertion
+			return errNoVerboseAssertion
```

Comment:
me, I'd check for this kind of thing as early as possible, before talking to the daemon at all. Not a blocker though.

Reply:
okay done, and updated the test to ensure no talking to the daemon happens in this case :zipper_mouth_face: